File: sdk-js/src/utils/validation.ts
/**
 * Validation utilities
 */

import { AgentrixValidationError } from './errors';

export function validateApiKey(apiKey: string): void {
  if (!apiKey || typeof apiKey !== 'string' || apiKey.trim().length === 0) {
    throw new AgentrixValidationError('API key is required');
  }
}

export function validateAmount(amount: number): void {
  if (typeof amount !== 'number' || amount <= 0) {
    throw new AgentrixValidationError('Amount must be a positive number');
  }
}

export function validateCurrency(currency: string): void {
  if (!currency || typeof currency !== 'string' || currency.length !== 3) {
    throw new AgentrixValidationError('Currency must be a 3-letter code (e.g., USD, CNY)');
  }
}

export function validatePaymentRequest(request: any): void {
  if (!request.amount) {
    throw new AgentrixValidationError('Amount is required');
  }
  validateAmount(request.amount);

  if (!request.currency) {
    throw new AgentrixValidationError('Currency is required');
  }
  validateCurrency(request.currency);

  if (!request.description) {
    throw new AgentrixValidationError('Description is required');
  }
}

File: sdk-python/agentrix/utils/validation.py
"""
Validation utilities
"""

from .errors import AgentrixValidationError


def validate_api_key(api_key: str) -> None:
    """Validate API key"""
    if not api_key or not isinstance(api_key, str) or not api_key.strip():
        raise AgentrixValidationError("API key is required")


def validate_amount(amount: float) -> None:
    """Validate amount"""
    if not isinstance(amount, (int, float)) or amount <= 0:
        raise AgentrixValidationError("Amount must be a positive number")


def validate_currency(currency: str) -> None:
    """Validate currency code"""
    if not currency or not isinstance(currency, str) or len(currency) != 3:
        raise AgentrixValidationError("Currency must be a 3-letter code (e.g., USD, CNY)")


def validate_payment_request(request: dict) -> None:
    """Validate payment request"""
    if "amount" not in request:
        raise AgentrixValidationError("Amount is required")
    validate_amount(request["amount"])

    if "currency" not in request:
        raise AgentrixValidationError("Currency is required")
    validate_currency(request["currency"])

    if "description" not in request:
        raise AgentrixValidationError("Description is required")

